content/docs/07-reference/01-ai-sdk-core/80-smooth-stream.mdx

```diff
@@ -42,14 +42,58 @@ const result = streamText({
     },
     {
       name: 'chunking',
-      type: '"word" | "line" | RegExp',
+      type: '"word" | "line" | RegExp | (buffer: string) => string | undefined | null',
       isOptional: true,
       description:
-        'Controls how the text is chunked for streaming. Use "word" to stream word by word (default), "line" to stream line by line, or provide a custom RegExp pattern for custom chunking.',
+        'Controls how the text is chunked for streaming. Use "word" to stream word by word (default), "line" to stream line by line, or provide a custom callback or RegExp pattern for custom chunking.',
     },
   ]}
 />
 
+#### Word chunking does not support non-latin languages
+
+> [!TIP]
+> The word based chunking **will not pair well** the following languages, that do not delimit words with spaces:
+>
+> - Chinese
+> - Japanese
+> - Vietnamese
+> - Thai
+> - Javanese (Aksara Jawa)
+>
+> For these languages, **it's recommended to either:**
+>
+> - **Recommended:** provide your own chunking regex/function that accounts for the language's word boundaries
+> - _Don't_ use smoothStream
```

Comment:
We have a `<Note>...</Note>` component that you can use instead.

Reply:
do i need to import it?

Reply:
looks like not

Reply:
done!